functions/consumer/redis-consumer/src/test/java/org/springframework/cloud/fn/consumer/redis/RedisConsumerQueueTests.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.fn.consumer.redis;
+
+import java.util.concurrent.TimeUnit;
+
+import org.junit.jupiter.api.Test;
+import org.springframework.data.redis.core.StringRedisTemplate;
+import org.springframework.messaging.Message;
+import org.springframework.messaging.support.MessageBuilder;
+import org.springframework.test.context.TestPropertySource;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ * @author Soby Chacko
+ */
+@TestPropertySource(properties = "redis.consumer.queue = test-queue")
+public class RedisConsumerQueueTests extends AbstractRedisConsumerTests {
+
+	@Test
+	public void testWithQueue() {
+		StringRedisTemplate redisTemplate = createStringRedisTemplate(redisConnectionFactory);
+
+		//Execute
+		Message<String> message = MessageBuilder.withPayload("hello").build();
+
+		redisConsumer.accept(message);
+
+		Object result = redisTemplate.boundListOps("test-queue").rightPop(5000, TimeUnit.MILLISECONDS);
+		assertThat(result).isNotNull();
+
+		assertThat(result).isEqualTo("hello");
```

Comment:
Just FYI: assertions can be combined or nested:
```
assertThat(result)
    .isNotNull()
    .isEqualTo("hello");
```
But in your case `.isNotNull()` is really redundant 😄